addons/point_of_sale/models/pos_session.py

```diff
@@ -646,7 +646,7 @@ def _create_picking_at_end_of_session(self):
             pickings.write({'pos_session_id': self.id, 'origin': self.name})
 
     def _create_balancing_line(self, data, balancing_account, amount_to_balance):
-        if (not float_is_zero(amount_to_balance, precision_rounding=self.currency_id.rounding)):
+        if (not float_is_zero(amount_to_balance, precision_rounding=self.company_id.currency_id.rounding)):
```

Comment:
```suggestion
        if not self.company_id.currency_id.is_zero(amount_to_balance):
```

Reply:
@smetl Thanks for your review.
I modified the code.